File: app/agents/fallback.py
"""FallbackAgent: erros, timeouts e limite de turnos. Sempre faz handoff.

Estático de propósito (nenhuma chamada de LLM) — precisa funcionar justamente
quando o LLM é o que está falhando.
"""

import logging
from typing import List

from app.agents.base import BaseAgent
from app.domain import AgentResult, ChatMessage
from app.messages import FALLBACK_HANDOFF

log = logging.getLogger("blip-agent.fallback")


class FallbackAgent(BaseAgent):
    source = "fallback"

    def system_prompt(self, user_message: str) -> str:  # nunca usado (execute é estático)
        return ""

    async def execute(self, user_message: str, history: List[ChatMessage]) -> AgentResult:
        log.info("FallbackAgent (history=%d) — handoff.", len(history))
        return AgentResult(
            response=FALLBACK_HANDOFF,
            should_handoff=True,
            handoff_reason="Encaminhado ao atendimento humano (limite de interações ou erro).",
            source="fallback",
        )

File: src/server/agents/fallback.ts
/**
 * FallbackAgent: erros, timeouts e limite de turnos. Sempre faz handoff.
 * Estático de propósito (sem LLM) — precisa funcionar quando o LLM é o que falha.
 * Porta `app/agents/fallback.py`.
 */

import { agentResult, type AgentResult, type ChatMessage } from "../domain";
import { getLogger } from "../logging";
import { FALLBACK_HANDOFF } from "../messages";
import { BaseAgent } from "./base";

const log = getLogger("blip-agent.fallback");

export class FallbackAgent extends BaseAgent {
  source = "fallback";

  systemPrompt(): string {
    return ""; // nunca usado (execute é estático)
  }

  async execute(_userMessage: string, history: ChatMessage[]): Promise<AgentResult> {
    log.info(`FallbackAgent (history=${history.length}) — handoff.`);
    return agentResult({
      response: FALLBACK_HANDOFF,
      shouldHandoff: true,
      handoffReason: "Encaminhado ao atendimento humano (limite de interações ou erro).",
      source: "fallback",
    });
  }
}
